components/camel-rabbitmq/src/main/java/org/apache/camel/component/rabbitmq/RabbitConsumer.java

```diff
@@ -160,12 +180,16 @@ public void stop() throws IOException, TimeoutException {
             channel.basicCancel(tag);
         }
         try {
+			lock.acquire();
             if (isChannelOpen()) {
                 channel.close();
             }
-        } catch (TimeoutException e) {
+            lock.release();
```

Comment:
same thing here... but it can go in the outer `finally` block, just after `catch (InterruptedException e1)`